src/main/java/org/opentripplanner/netex/loader/parser/ServiceCalendarFrameParser.java

```diff
@@ -84,13 +85,29 @@ private void parseDayType(JAXBElement<?> dt) {
     }
   }
 
-  private void parseOperatingPeriods(OperatingPeriodsInFrame_RelStructure element) {
-    if (element == null) {
+  private void parseOperatingPeriods(OperatingPeriodsInFrame_RelStructure operatingPeriods) {
+    if (operatingPeriods == null) {
+      return;
+    }
+
+    for (OperatingPeriod_VersionStructure p : operatingPeriods.getOperatingPeriodOrUicOperatingPeriod()) {
+      parseOperatingPeriod(p);
+    }
+  }
+
+  private void parseOperatingPeriods(OperatingPeriods_RelStructure operatingPeriods) {
+    if (operatingPeriods == null) {
       return;
     }
 
-    for (OperatingPeriod_VersionStructure p : element.getOperatingPeriodOrUicOperatingPeriod()) {
-      operatingPeriods.add((OperatingPeriod) p);
+    for (Object p : operatingPeriods.getOperatingPeriodRefOrOperatingPeriodOrUicOperatingPeriod()) {
+      parseOperatingPeriod(p);
+    }
+  }
+
+  private void parseOperatingPeriod(Object operatingPeriod) {
+    if (operatingPeriod instanceof OperatingPeriod_VersionStructure op) {
+      operatingPeriods.add(op);
     }
```

Comment:
We typically call `NetexParser#warnOnMissingMapping` for all entities which are not mapped, could we add it also here for other types of possible values?